src/txHashBuilder.c

```diff
@@ -658,11 +807,14 @@ void txHashBuilder_addWithdrawal(
         uint64_t amount
 )
 {
+	_TRACE("state = %u, remainingWithdrawals = %u", builder->state, builder->remainingWithdrawals);
+
 	ASSERT(builder->state == TX_HASH_BUILDER_IN_WITHDRAWALS);
 	ASSERT(builder->remainingWithdrawals > 0);
-	ASSERT(rewardAddressSize == 1 + ADDRESS_KEY_HASH_LENGTH);
 	builder->remainingWithdrawals--;
 
+	ASSERT(rewardAddressSize == 1 + ADDRESS_KEY_HASH_LENGTH);
```

Comment:
seems like a redundant change

Reply:
After this change, argument checking is separated from builder state checking. So the logic is "trace the state of the builder, assert the state is consistent and that we are supposed to call this function at all, assert that the given arguments are correct etc.".